returnn/config.py

```diff
@@ -209,13 +269,17 @@ def check_consistency(self):
             assert key not in self.post_config, (
                 "%s in post_config would overwrite existing entry in config" % key
             )
+        assert bool(self.staged_network_dict) != bool("network" in self.config)
```

Comment:
At some point we would also like to cater for custom `get_network` functions (compare #95) . But this has time for another PR.